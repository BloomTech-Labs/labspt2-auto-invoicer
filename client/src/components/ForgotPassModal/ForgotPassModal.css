.password-form {
  width: 300px;
  margin: 0 auto;
  min-height: 150px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.password-btn {
  margin: 0 10px;
  font-size: 2rem;
  padding: 5px 2px;
  border: 1px solid black;
  border-radius: 5px;
}
