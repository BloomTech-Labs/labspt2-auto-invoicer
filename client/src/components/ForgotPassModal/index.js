import React from "react";
import PropTypes from "prop-types";

// import components here
import Modal from "../reusableComponents/Modal";

// import css here
import "./ForgotPassModal.css";

class ForgotPassModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: ""
    };
  }
  changeHandler = e => {
    return this.setState({ [e.target.name]: e.target.value });
  };
  createResetObject = e => {
    e.preventDefault();
    const reset = this.state.email;
    this.props.passwordReset({ email: reset });
    this.props.click()
  };
  render() {
    return (
      <Modal close={this.props.click} className="password-container">
        <h1 className="modal-title">Reset Password</h1>
        <form onSubmit={this.createResetObject} className="password-form">
          <input
            className="signin-input"
            type="email"
            placeholder="email"
            required
            name="email"
            onChange={this.changeHandler}
            value={this.state.email}
          />
          <button className="password-btn" type="submit">
            Reset password
          </button>
        </form>
      </Modal>
    );
  }
}

ForgotPassModal.propTypes = {
  click: PropTypes.func.isRequired
};

export default ForgotPassModal;
