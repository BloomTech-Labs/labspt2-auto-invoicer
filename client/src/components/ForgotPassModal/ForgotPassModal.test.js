import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import ForgotPassModal from "../ForgotPassModal";

describe("<ForgotPassModal />", () => {
  it("renders Forgot Password Modal", () => {
    const { asFragment } = render(<ForgotPassModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
