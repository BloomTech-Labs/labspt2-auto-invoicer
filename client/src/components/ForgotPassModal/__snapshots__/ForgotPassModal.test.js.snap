// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassModal /> renders Forgot Password Modal 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <span
        class="close-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1"
          focusable="false"
          id="close"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <h1
        class="modal-title"
      >
        Reset Password
      </h1>
      <form
        class="password-form"
      >
        <input
          class="signin-input"
          name="email"
          placeholder="email"
          required=""
          type="email"
          value=""
        />
        <button
          class="password-btn"
          type="submit"
        >
          Reset password
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
